File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/java/sniker1223.java
import java.util.Arrays;
import java.util.List;

public class sniker1223 {

  public static void main(String[] args) {
    int i = 10;
    int x = 12;
    String name = "Sara";

    // Arithmetic Operators
    System.out.println("Arithmetic Operators");
    System.out.println("+ Addition: " + (1 + 2));
    System.out.println("- Substraction: " + (1 - 2));
    System.out.println("* Multiplication: " + (1 * 2));
    System.out.println("/ Division: " + (1 / 2));
    System.out.println("% Modulus: " + (1 % 2));
    System.out.println("++ Increment: " + (++i));
    System.out.println("-- Decrement: " + (--i));

    // Assignment Operators
    System.out.println("\nAssignment Operators");
    System.out.println("= Assigment: i = " + i);
    System.out.println("+= Addition assigment: i += " + (i += 2));
    System.out.println("-= Substraction assigment: i -= " + (i -= 2));
    System.out.println("*= Multiplication assigment: i *= " + (i *= 2));
    System.out.println("/= Division assigment: i /= " + (i /= 2));
    System.out.println("%= Modulus assigment: i %= " + (i %= 7));
    System.out.println("&= Bitwise AND assigment: i &= " + (i &= 12));
    System.out.println("|= Bitwise OR assigment: i |= " + (i |= 4));
    System.out.println("^= Bitwise XOR assigment: i ^= " + (i ^= 8));
    System.out.println(">>= right shift assigment: i >>= " + (i >>= 1));
    System.out.println("<<= left shift assigment: i <<= " + (i <<= 1));

    // Comparison Operators
    System.out.println("\nComparison Operators");
    System.out.println("> Greater than: " + i + "  > " + x + " = " + (i > x));
    System.out.println("> Less than: " + i + "  < " + x + " = " + (i < x));
    System.out.println("== Equal to: " + i + "  == " + x + " = " + (i == x));
    System.out.println("!= Equal to: " + i + "  != " + x + " = " + (i != x));
    System.out.println(">= Greater than: " + i + "  >= " + x + " = " + (i >= x));
    System.out.println("<= Less than: " + i + "  <= " + x + " = " + (i <= x));

    // Logical Operators
    System.out.println("\nLogical Operators");
    System.out.println("&& Logical AND: " + x + " < 5 && " + x + " < 10" + " = " + (x < 5 && x < 10));
    System.out.println("||  Logical OR: " + x + " < 5 || " + x + " < 20" + " = " + (x < 5 || x < 20));
    System.out.println("!  Logical NOT: !(" + x + " < 5 && " + x + " < 10)" + " = " + !(x < 5 || x < 10));

    // Miscellaneous Operators
    System.out.println("\nMiscellaneous Operators");
    System.out.println("?: Ternary operator: (" + i + " == 1 ? 20 : 30 value of i is : " + ((i == 1) ? 20 : 30));
    System.out.println("instanceof Instanceof operator: " + name + " instanceof String  = " + (name instanceof String));

    // CONTROL STRUCTURES
    System.out.println("\nCONTROL STRUCTURES");
    // If/Else/Else If
    System.out.println("If/Else/Else If");
    if (i > 2) {
      System.out.println(i + " is higher than 2");
    } else {
      System.out.println(i + " is lower or equal than 2");
    }
    // Switch
    System.out.println("\nSwitch");
    i = 3;
    switch (i) {
      case 0:
        System.out.println(i + " is equal to 0");
        break;
      case 1:
        System.out.println(i + " is equal to 1");
        break;
      default:
        System.out.println(i + " is either negative, or higher than 1");
        break;
    }

    // Loops
    System.out.println("\nLoops");
    System.out.println("for");
    for (int a = 1; a <= 10; a++) {
      System.out.println(a);
    }
    System.out.println("\nwhile");
    while (i <= 10) {
      System.out.println(i);
      i++;
    }

    // Break with foreach
    System.out.println("\nBreak");
    List<String> names = Arrays.asList("Sara", "Selena");
    String nameBreak = "Sara";
    for (String cadena : names) {
      if (cadena.equals(nameBreak)) {
        System.out.println(cadena);
        break;
      }

    }

    // Challenge. Print numbers between 10 and 55(included), even and odd but not
    // print 16 and multiple of 3.
    int n;
    for (n = 0; n <= 55; n++) {
      if (n >= 10 && n % 2 == 0 && n != 16 && !(n % 3 == 0) || n == 55) {
        System.out.println(n);
      }
    }
  }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/sniker1223.py
# Arithmetic Operators
print('Arithmetic Operators')
print('+ Addition: ',5+3)
print('- Subtraction: ',5-3)
print('* Multiplication: ',5*3)
print('/ Division: ',5/3)
print('% Modulus: ',5%3)
print('** Exponentiation: ',5**3)
print('// Floor division: ',5//3)

# Assignment Operators
print('\nAssignment Operators')
let =10
print('= Assigment: ',let)
let+=1
print('+= Addition assigment: ',let)
let-=1
print('-= Substraction assigment: ',let)
let *=10
print('*= Multiplication assigment: ',let)
let /=5
print('/= Division assigment: ',let)
let %=1.5
print('%= Modulus assigment: ',let)
var =5
var &=3
print('&= Bitwise AND assigment: ',var)
var |=3
print('|= Bitwise OR assigment: ',var)
var ^=7
print('^= Bitwise XOR assigment: ',var)
var >>=0
print('>>= right shift assigment: ',var)
var <<=1
print('<<= left shift assigment: ',var)

# Comparison Operators
print('\nComparison Operators')
print('> Greater than:',1 > 2)
print('> Less than:',1 < 2)
print('== Equal to:',1 == 2)
print('!= Equal to:',1 != 2)
print('>= Greater to:',1 >= 2)
print('<= Less to:',1 <= 2)

# Logical Operators
print('\nLogical Operators')
x = 2
print('and',x < 5 and  x < 10)
print('or',x < 5 or  x < 10)
print('not',not(x < 5 and x < 10))

# Identity Operators
print('is', 1 is 1)
print('is not', 1 is not 1)

# Membership Operators
fuits = ["apple", "banana"]
print("banana" in fuits)
print("pineapple" not in fuits)

# CONTROL STRUCTURES
print('\nCONTROL STRUCTURES')
print('if')
a = 33
b = 200
if b > a:
  print("b is greater than a")

print('\nelif')
a = 33
b = 33
if b > a:
  print("b is greater than a")
elif a == b:
  print("a and b are equal")

print('\nelse')
a = 200
b = 33
if b > a:
  print("b is greater than a")
elif a == b:
  print("a and b are equal")
else:
  print("a is greater than b")

print('\nShort Hand If')
a = 200
b = 33
if a > b: print("a is greater than b")

print('\nShort Hand If ... Else')
a = 2
b = 330
print("A") if a > b else print("B")

# Loops
print('\nwhile')
i = 1
while i < 6:
  print(i)
  i += 1

print('\nwhile with break statement')
i = 1
while i < 6:
  print(i)
  if i == 3:
    break
  i += 1

print('\nwhile with continue statement')
i = 0
while i < 6:
  i += 1
  if i == 3:
    continue
  print(i)

print('\nwhile with else Statement')
i = 1
while i < 6:
  print(i)
  i += 1
else:
  print("i is no longer less than 6")

print('\nfor')
fruits = ["apple", "banana", "cherry"]
for x in fruits:
  print(x) 

print('\nfor through a String')

for x in "banana":
  print(x)

print('\nfor with break Statement')
fruits = ["apple", "banana", "cherry"]
for x in fruits:
  print(x)
  if x == "banana":
    break

print('\nfor with continue Statement')
fruits = ["apple", "banana", "cherry"]
for x in fruits:
  if x == "banana":
    continue
  print(x) 

print('\nfor with range() Function')
for x in range(6):
  print(x) 
  
# values from 2 to 6
for x in range(2, 6):
  print(x)

print('\nElse in For Loop')
for x in range(6):
  print(x)
else:
  print("Finally finished!")

# Challenge. Print numbers between 10 and 55(included), even and odd but not
# print 16 and multiple of 3.
print('\nchallenge')
for x in range(10, 56):
  if x >= 10 and x % 2 == 0 and x != 16 and not x % 3 == 0 or x == 55: 
    print(x)